client/src/components/Libraries/LibraryFolder/TopToolbar/FolderTopBar.vue

```diff
@@ -1,52 +1,326 @@
+
+            const selected = await services.getFilteredFolderContents(
+                props.folderId,
+                props.unselected,
+                props.searchText
+            );
+
+            emit("setBusy", false);
+
+            return selected;
+        } catch (err) {
+            console.error(err);
+        } finally {
+            emit("setBusy", false);
+        }
+
+        return [];
+    } else {
+        return props.selected;
+    }
+}
+
+async function deleteSelected() {
+    try {
+        const selected = await getSelected();
+
+        deleteSelectedItems(
+            selected,
+            (deletedItem: any) => emit("deleteFromTable", deletedItem),
+            () => emit("refreshTable"),
+            () => emit("refreshTableContent")
+        );
+    } catch (err) {
+        console.error(err);
+    }
+}
+
+async function findCheckedItems(idOnly = true) {
+    const folders: any[] = [];
+    const datasets: any[] = [];
+
+    try {
+        const selected = await getSelected();
+
+        selected.forEach((item: any) => {
+            const selectedItem = idOnly ? item.id : item;
+
+            item.type === "file" ? datasets.push(selectedItem) : folders.push(selectedItem);
+        });
+    } catch (err) {
+        console.error(err);
+    }
+
+    return { datasets, folders };
+}
+
+async function downloadData(format: string) {
+    try {
+        const { datasets, folders } = await findCheckedItems();
+
+        if (props.selected.length === 0) {
+            Toast.info("You must select at least one dataset to download");
+
+            return;
+        }
+
+        download(format, datasets, folders);
+    } catch (err) {
+        console.error(err);
+    }
+}
+
+async function importToHistoryModal(isCollection: boolean) {
+    try {
+        const { datasets, folders } = await findCheckedItems(!isCollection);
+
+        const checkedItems: any = props.selected;
+
+        checkedItems.dataset_ids = datasets;
+        checkedItems.folder_ids = folders;
+
+        if (isCollection) {
+            new mod_import_collection.ImportCollectionModal({
+                selected: checkedItems,
+            });
+        } else {
+            new mod_import_dataset.ImportDatasetModal({
+                selected: checkedItems,
+            });
+        }
+    } catch (err) {
+        console.error(err);
+    }
+}
+
+// TODO: after replacing the selection dialog with the new component that is not using jquery
+function addDatasets(source: string) {
+    new mod_add_datasets.AddDatasets({
+        source: source,
+        id: props.folderId,
+        updateContent: updateContent,
+        list_genomes: genomesList.value,
+        list_extensions: extensionsList.value,
+    });
+}
+
+function updateContent() {
+    emit("fetchFolderContents");
+}
+// END TODO
+
+async function fetchExtAndGenomes() {
+    try {
+        const { data: dataTypes } = await fetchDatatypes({ extension_only: false });
```

Comment:
Consider refactoring this to use the existing composable `useDetailedDatatypes` instead of fetching the datatypes every time.

The same can be done for "genomes" (which should be dbKeys), but I'm not sure if we already have a composable or store for them.

Reply:
Thanks for the information, @davelopez! I applied the necessary changes and pushed them. The `datatypes` from `useDetailedDatatypes` has a different format than this component needed and the selector (`select2`) is now broken but will be replaced in the next commits.